### PR TITLE
GraveRobber fixes, Infection and H2X Cleanup

### PR DESCRIPTION
**Pull request by Berthalamew**

### CODE REVIEW

File: xlive/H2MOD/Variants/H2X/H2X.cpp [hunk 1]
```diff
@@ -43,91 +42,47 @@ void H2X::ApplyMapLoadPatches(bool enable)
...
+		{
+			s_weapon_group_definition* weapon_tag = tags::get_tag_fast<s_weapon_group_definition>(weapon_datum);
+			weapon.rounds_per_second_based ?
+				weapon_tag->barrels[weapon.barrel_data_block_index]->rounds_per_second_upper = (enable ? weapon.h2x_rate_of_fire : weapon.original_rate_of_fire)
+				: weapon_tag->barrels[weapon.barrel_data_block_index]->fire_recovery_time = (enable ? weapon.h2x_rate_of_fire : weapon.original_rate_of_fire);
 		}
 	}
 
-	if (enable && !Memory::IsDedicatedServer() && h2mod->GetEngineType() == e_engine_type::_multiplayer)
+	if (enable && !Memory::IsDedicatedServer() && s_game_globals::game_is_multiplayer())
```
Comment (nukeulater) [inline xlive/H2MOD/Variants/H2X/H2X.cpp]: `h2mod->GetEngineType() == e_engine_type::_multiplayer` was right, you are breaking this, also for the code bellow you should add a size check for the sound class blocks if possible

File: xlive/Project_Cartographer.vcxproj.user [hunk 2]
```diff
@@ -7,6 +7,13 @@
     <DebuggerFlavor>WindowsLocalDebugger</DebuggerFlavor>
     <LocalDebuggerCommandArguments>-nointro -windowed</LocalDebuggerCommandArguments>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='GameRewriteDebug|Win32'">
```
Comment (nukeulater) [inline xlive/Project_Cartographer.vcxproj.user]: please don't add this to the main repository

File: xlive/Project_Cartographer.vcxproj.user [hunk 3]
```diff
@@ -22,6 +29,13 @@
     <LocalDebuggerDebuggerType>NativeOnly</LocalDebuggerDebuggerType>
     <LocalDebuggerCommandArguments>-nointro  -novsync</LocalDebuggerCommandArguments>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='GameRewriteRelease|Win32'">
```
Comment (nukeulater) [inline xlive/Project_Cartographer.vcxproj.user]: same with this, don't add to the main repository

File: xlive/H2MOD/Variants/GraveRobber/GraveRobber.cpp [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+typedef void(__thiscall* update_player_score_t)(void* thisptr, unsigned short a2, int a3, int a4, int a5, char a6);
+extern update_player_score_t p_update_player_score;
+
+void GraveRobber::PickupSkull(datum playerIdx, datum skullDatum)
+{
+	const short absPlayerIdx = DATUM_INDEX_TO_ABSOLUTE_INDEX(playerIdx);
+
+	if (!DATUM_IS_NONE(skullDatum))
+	{
+		typedef char* (__cdecl* get_score_data_ptr)();
```
Comment (nukeulater) [inline xlive/H2MOD/Variants/GraveRobber/GraveRobber.cpp]: `get_score_data` is enough, also the type definition should use `_t` at the end

File: xlive/H2MOD/Variants/GraveRobber/GraveRobber.cpp [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+		if (player_score_data)
+		{
+			if (!s_game_globals::game_is_predicted())
+			{
+				p_update_player_score(player_score_data, absPlayerIdx, 0, 1, -1, 0);
+			}
+			HaloScript::ObjectDestroy(skullDatum);
+
+			if (DATUM_INDEX_TO_ABSOLUTE_INDEX(h2mod->get_player_datum_index_from_controller_index(0)) == absPlayerIdx ||
+				DATUM_INDEX_TO_ABSOLUTE_INDEX(h2mod->get_player_datum_index_from_controller_index(1)) == absPlayerIdx ||
```
Comment (nukeulater) [inline xlive/H2MOD/Variants/GraveRobber/GraveRobber.cpp]: we should create a helper function for this, also should be a loop

File: xlive/H2MOD/Variants/Infection/Infection.h [hunk 6]
```diff
@@ -22,31 +55,20 @@ class Infection : public ICustomGameVariant
...
+	virtual bool OnAutoPickupHandler(ExecTime execTime, datum playerIdx, datum objectIdx) override { return false; }
+	virtual bool OnPlayerScore(ExecTime execTime, void* thisptr, unsigned short a2, int a3, int a4, int a5, char a6) override { return false; }
 
-	virtual bool OnPlayerScore(ExecTime execTime, void* thisptr, unsigned short a2, int a3, int a4, int a5, char a6) override
-	{
-		return false;
-	}
-
-	Infection();
-	~Infection();
```
Comment (nukeulater) [inline xlive/H2MOD/Variants/Infection/Infection.h:42]: deconstructor should be virtual here too
otherwise good catch

**Review by nukeulater — CHANGES_REQUESTED** [review]

File: xlive/Project_Cartographer.vcxproj.user (hunk 2 above)
Comment (Berthalamew) [reply xlive/Project_Cartographer.vcxproj.user]: https://github.com/pnill/cartographer/pull/407/commits/15ddad208effb18819e21dedf764dd9bbc93430a#diff-c60806c8d16d9111955b303c7360c44628dff63caab58e858e8c86a84cf0c878L10-L16
Fixed

File: xlive/Project_Cartographer.vcxproj.user (hunk 3 above)
Comment (Berthalamew) [reply xlive/Project_Cartographer.vcxproj.user]: https://github.com/pnill/cartographer/pull/407/commits/15ddad208effb18819e21dedf764dd9bbc93430a#diff-c60806c8d16d9111955b303c7360c44628dff63caab58e858e8c86a84cf0c878L32-L38
Fixed

File: xlive/H2MOD/Variants/H2X/H2X.cpp (hunk 1 above)
Comment (Berthalamew) [reply xlive/H2MOD/Variants/H2X/H2X.cpp]: https://github.com/pnill/cartographer/pull/407/commits/15ddad208effb18819e21dedf764dd9bbc93430a#diff-53f3f1a9f579c86f48f5e72e7355f36eddde2a46ee2fed9289c1410426dba388R56
Fixed

File: xlive/H2MOD/Variants/GraveRobber/GraveRobber.cpp (hunk 4 above)
Comment (Berthalamew) [reply xlive/H2MOD/Variants/GraveRobber/GraveRobber.cpp]: https://github.com/pnill/cartographer/pull/407/commits/15ddad208effb18819e21dedf764dd9bbc93430a#diff-ac5bc43b1a79579fc36772ba42d52a8f5a4cd2c045d586ce2c21b4528b0e5d79R69-R70
Changed it to get_score_data_t

File: xlive/H2MOD/Variants/Infection/Infection.h (hunk 6 above)
Comment (Berthalamew) [reply xlive/H2MOD/Variants/Infection/Infection.h:42]: https://github.com/pnill/cartographer/pull/407/commits/15ddad208effb18819e21dedf764dd9bbc93430a#diff-722acf67b979a805dd7e77189f0f0f4ee9e23aff90c6145b6c05ab85371695b1R65
Changed it to virtual

File: xlive/H2MOD/Variants/GraveRobber/GraveRobber.cpp (hunk 5 above)
Comment (Berthalamew) [reply xlive/H2MOD/Variants/GraveRobber/GraveRobber.cpp]: https://github.com/pnill/cartographer/pull/407/commits/15ddad208effb18819e21dedf764dd9bbc93430a#diff-ac5bc43b1a79579fc36772ba42d52a8f5a4cd2c045d586ce2c21b4528b0e5d79
Changed it to a loop, and reworked the function so it doesn't have a bunch of nested if statements

**Pull request merged**